doc/conf.py

```diff
@@ -30,6 +30,11 @@
 # General configuration
 # ---------------------
 
+# Unless we catch the warning explicitly somewhere, a warning should cause the
+# docs build to fail. This is especially useful for getting rid of deprecated
+# usage in the gallery.
+warnings.filterwarnings('error', append=True)
```

Comment:
```suggestion
warnings.filterwarnings('error', append=True)
```
How does this affect warnings generated by sphinx? We often have multiple warnings due to incorrect ReST syntax/wrong cross-references, and I don't want to fail on them individually.

Reply:
No Sphinx will still generate each docstring individually and report all the errors at the end in a single dump. 

AFAICT, in order to short-circuit the documentation building you'd have to actually trigger a proper bug in Sphinx. 

I tested by inserting various invalid text roles `:asdflkj:``stuff`` ` in multiple files and a single `make html` does in fact show warnings in one go.

Reply:
That said, making sphinx_gallery `abort_on_error` does have this property, so I've reverted that change.

With the new warnings, any warning or error will cause CI to fail, however every warning and error will be shown in the output if there are multiple, both for gallery examples/tutorials and for API docs.